rav1e_js/src/encoder.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2020, The rav1e contributors. All rights reserved
+//
+// This source code is subject to the terms of the BSD 2 Clause License and
+// the Alliance for Open Media Patent License 1.0. If the BSD 2 Clause License
+// was not distributed with this source code in the LICENSE file, you can
+// obtain it at www.aomedia.org/license/software. If the Alliance for Open
+// Media Patent License 1.0 was not distributed with this source code in the
+// PATENTS file, you can obtain it at www.aomedia.org/license/patent.
+
+#![allow(non_snake_case)]
+
+use rav1e::prelude::{Config, Context, EncoderStatus};
+
+use wasm_bindgen::prelude::*;
+
+use crate::utils::construct_js_err;
+use crate::{EncoderConfig, Frame, Packet};
+
+/// Wrapper around the encoder context (`rav1e::api::context::Context<u16>`).
+///
+/// Contains the encoding state.
+#[wasm_bindgen]
+pub struct Encoder {
+  ctx: Context<u16>,
+}
+
+#[wasm_bindgen]
+impl Encoder {
+  #[wasm_bindgen(constructor)]
+  pub fn from_config(conf: EncoderConfig) -> Result<Encoder, JsValue> {
+    let cfg = Config::new().with_encoder_config(conf.conf);
+
+    match cfg.new_context() {
+      Ok(c) => Ok(Encoder { ctx: c }),
+      Err(e) => Err(construct_js_err(e, "Invalid EncoderConfig")),
+    }
+  }
+
+  pub fn default() -> Result<Encoder, JsValue> {
+    Self::from_config(EncoderConfig::new())
+  }
+
+  pub fn debug(&self) -> String {
+    format!("{:?}", self.ctx)
+  }
+
+  /// Allocates and returns a new frame.
+  pub fn newFrame(&self) -> Frame {
```

Comment:
Either use `underscore_separator()` or `capitalizedSeparator()` style.

Reply:
I will go with camelCase (capitalizedSeparator), because it is the js-standart